### ISSUE TITLE
`space-unary-ops`: allow different overrides for `!` (not operator) and `!` (TypeScript non-null assertion)

### ISSUE THREAD

**Issue by stefanvdlugt** [eslint-stylistic/eslint-stylistic#1117]
### Clear and concise description of the problem

In the `space-unary-ops` rule we'd like to be able to treat the `!` (NOT) and `!` (TS non-null assertion) differently, so that this becomes accepted:

```typescript
// we know that myObject is not null here
if (! myObject!->prop === 3) {
    console.log(...);
}
```

### Suggested solution

Add an option `ts-non-null` to the `overrides` object. These overrides would cause the above example to be accepted.

```typescript
{
    '!': false,
    'ts-non-null': true,
}
```

For backwards compatibility, we could let the behavior of the non-null assertion operator be determined in this order:

1. `overrides['ts-non-null']`
2. `overrides['!']`
3. `nonwords`

### Alternative

- Instead of overrides accepting only `true` or `false`, also allow options `'only-before'` and `'only-after'`.

### Additional context

Happy to provide a pull request, but need some input on which approach would work best here.

### Validations

- [x] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [x] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [x] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

### Contributes

- [x] If this feature request is accepted, I am willing to submit a PR to fix this issue

**Comment by 9romise** [comment]
If needed, instead of `only-*` words, I'd like `{ before: true, after: true }` for consistency.
